### PR TITLE
telemetry(featuredev): add feedback telemetry

### PR DESCRIPTION
**Pull request by chengoramazon**
## Problem

- feature dev is missing feedback telemetry. 
feedback

## Solution

- Add feedback telemetry

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazonq/.changes/next-release/Feature-fb724754-850b-41b3-b3dd-2484b40720ff.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Amazon Q feature dev: add feedback telemetry."
+}
```
Comment (hayemaxi) [inline packages/amazonq/.changes/next-release/Feature-fb724754-850b-41b3-b3dd-2484b40720ff.json:4]:
```suggestion
{
	"type": "Feature",
	"description": "Amazon Q feature dev: add feedback telemetry."
}
```
Telemetry isn't user-facing, so we don't need a changelog item. But thanks for being pro-active about adding one!

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 2]
```diff
@@ -202,6 +209,21 @@ export class FeatureDevController {
         }
     }
 
+    private async processChatItemFeedbackMessage(message: any) {
```
Comment (hayemaxi) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:212]: Is there no type for message? If yes, let's use it. If not, why?

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 3]
```diff
@@ -202,6 +209,21 @@ export class FeatureDevController {
...
+
+        await globals.telemetry.postFeedback({
+            comment: `${JSON.stringify({
+                type: 'featuredev-chat-answer-feedback',
+                conversationId: session?.conversationId ?? '',
+                messageId: message?.messageId,
+                reason: message?.selectedOption,
+                userComment: message?.comment,
+            })}`,
+            sentiment: 'Negative',
```
Comment (hayemaxi) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:223]: Can it only ever be negative?

Comment (chengoramazon) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:223]: Yes, Mynah UI only allows users report feedback when they click on `thumb down ` in the message, so it is always Negative.

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts (hunk 2 above)
Comment (chengoramazon) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:212]: No, All messages are `any` in the controller, I will add an action item in the backlog.

**Review by willyyhuang — APPROVED** [review]

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts (hunk 3 above)
Comment (hayemaxi) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:223]: Ok. Let's add a comment:

```suggestion
            sentiment: 'Negative', // The chat UI reports only negative feedback currently.
```

**Review by hayemaxi — APPROVED** [review]

File: packages/core/src/amazonqFeatureDev/app.ts [hunk 4]
```diff
@@ -26,6 +26,7 @@ export function init(appContext: AmazonQAppInitContext) {
         followUpClicked: new vscode.EventEmitter<any>(),
         openDiff: new vscode.EventEmitter<any>(),
         processChatItemVotedMessage: new vscode.EventEmitter<any>(),
```
Comment (jpinkney-aws) [inline packages/core/src/amazonqFeatureDev/app.ts:28]: How is this new code different from processChatItemVotedMessage?

**Review by willyyhuang — APPROVED** [review]

**Comment by jpinkney-aws** [comment]
Can you clarify in your PR description what activates this telemetry? Is it the thumbs up/down button or something else?

File: packages/core/src/amazonqFeatureDev/app.ts (hunk 4 above)
Comment (chengoramazon) [reply packages/core/src/amazonqFeatureDev/app.ts:28]: It is used to record more details when users click on thumb down. Added a screenshot in the description.

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**